tests/model_registry/conftest.py

```diff
@@ -303,3 +316,61 @@ def generated_schema(model_registry_instance_rest_endpoint: str) -> Any:
         uri="https://raw.githubusercontent.com/kubeflow/model-registry/main/api/openapi/model-registry.yaml",
         base_url=f"https://{model_registry_instance_rest_endpoint}/",
     )
+
+
+def pytest_sessionstart(session: pytest.Session) -> None:
+    """Downloads external tests into the local project structure before test execution."""
+    if session.config.option.model_registry_upstream:
+        LOGGER.info("pytest_sessionstart is running")
+        os.makedirs(UPSTREAM_TESTS_DIR, exist_ok=True)
+        for file in FILES:
+            os.system(f"curl -o {UPSTREAM_TESTS_DIR}/{file} {REPO_API_URL}/{FILE_PATH}/{file}")
+            LOGGER.info(f"Downloaded {file} into {UPSTREAM_TESTS_DIR}")
+
+        # Ensure pytest can discover the external tests
+        sys.path.insert(0, UPSTREAM_TESTS_DIR)
+    else:
+        LOGGER.info("Skipping upstream test fetching for Model registry. Enable with --model-registry-upstream=True")
+
+
+def pytest_collection_modifyitems(session: pytest.Session, config: pytest.Config, items: pytest.Item) -> None:
+    """Ensures pytest detects new test modules within the cloned repository."""
```

Comment:
the download is before item collection, do you know why is this needed? pytest should be able to collect the tests

Reply:
You're right, it should work without it as well and it does in collect-only mode. I'll run a full test without it and update if everything checks out

Reply:
Done